### PR TITLE
New Gradients Database

### PR DESCRIPTION
**Pull request by bbenetskyy**
//todo add details when PR will be ready

# This is Draft Please do not Review it & Do not merge it

### CODE REVIEW

File: Playground/Playground.Android/Playground.Android.csproj [hunk 1]
```diff
@@ -106,4 +106,9 @@
     </ProjectReference>
   </ItemGroup>
   <Import Project="$(MSBuildExtensionsPath)\Xamarin\Android\Xamarin.Android.CSharp.targets" />
+  <ProjectExtensions>
+    <VisualStudio>
+      <UserProperties XamarinHotReloadDebuggerTimeoutExceptionPlaygroundAndroidHideInfoBar="True" />
```
Comment (bbenetskyy) [inline Playground/Playground.Android/Playground.Android.csproj:111]: Not Sure If we need it, I will try to remove it and check project

File: Playground/Playground.Data/Playground.Data.csproj [hunk 2]
```diff
@@ -33,4 +78,8 @@
     <PackageReference Include="Xamarin.Forms" Version="4.6.0.800" />
   </ItemGroup>
 
+  <ItemGroup>
+    <EmbeddedResource Include="Resources\SimplePatterns.json" />
```
Comment (bbenetskyy) [inline Playground/Playground.Data/Playground.Data.csproj:82]: hmm, strange VS works, I will move it manually to an item group above

File: Playground/Playground.Data/Resources/Metadata.json [hunk 3]
```diff
@@ -1,14 +1,37 @@
 {
-  "version": 1,
+  "version": 33,
```
Comment (bbenetskyy) [inline Playground/Playground.Data/Resources/Metadata.json:2]: When I update Popular Category, I will change this version to `2`

File: Playground/Playground/Services/GalleryService.cs [hunk 4]
```diff
@@ -19,7 +19,9 @@ public GalleryService()
 
         public IEnumerable<GradientItem> GetGradients(string tag)
         {
-            return _gradientRepository.GetByTag(tag).Select(MapGradient);
+            var x = _gradientRepository.GetByTag(tag);
```
Comment (bbenetskyy) [inline Playground/Playground/Services/GalleryService.cs:22]: this is tmp Debug changes, I should remove it

File: Playground/Playground/Views/GalleryCategoriesPage.xaml [hunk 5]
```diff
@@ -1,28 +1,39 @@
...
+    xmlns:models="clr-namespace:Playground.Models"
+    Title="Gradient Gallery"
+    BindingContext="{Binding GalleryCategoriesViewModel, Source={StaticResource ViewModelLocator}, Mode=OneTime}">
     <Grid Padding="5">
-        <CollectionView x:Name="CategoriesList" 
-                        ItemsSource="{Binding Categories}" 
-                        SelectionMode="Single"
-                        SelectedItem="{Binding SelectedItem}" 
-                        SelectionChanged="GalleryList_OnSelectionChanged">
+        <CollectionView
```
Comment (bbenetskyy) [inline Playground/Playground/Views/GalleryCategoriesPage.xaml:11]: This whole file looks like don't have any changes - so it's should be reverted

**Comment by bbenetskyy** [comment]
@mgierlasinski - you may do your review, but while I'm not fixing all my comments from own review - pls DO NOT MERGE IT 👾😋

**Comment by mgierlasinski** [comment]
We have to make sure we will not update Playground with broken gradients, maybe there are more issues that we could fix on css parser side. As for performance, yes I did notice that also, loader is not an option, this is at rendering level. We should try incremental loading for CollectionView.

**Comment by bbenetskyy** [comment]
@mgierlasinski after your review you can immediately merge it. After merging proper issues for fix Complex Patterns and others will be created by me(or you)

**Pull request merged**